source/classes.tex

```diff
@@ -6184,8 +6184,10 @@
 In the following copy-initialization contexts,
 a move operation is first considered before attempting a copy operation:
 \begin{itemize}
-\item If the \grammarterm{expression} in a \tcode{return}\iref{stmt.return} or
-\tcode{co_return}\iref{stmt.return.coroutine} statement
+\item If the \grammarterm{expression}
```

Comment:
No.

p1 talks about copy/move elision, which is totally different.

p3 talks about trying a move under certain circumstances; p3 starts with the definition of implicitly movable entity. Then, it says "In the following situations, a move is considered before attempting a copy". If e.g. the "return" is returning a global variable, this is not one of those situations, and we never try a move. We always try a copy, though.

Reply:
Reworded to cover both functions and lambda-expressions.

Reply:
If we phrase it as the following, it seems more clear  
> if the expression or operand is an implicitly movable entity, overload resolution to select the constructor for the copy or the return_­value overload to call is first performed as if the expression or operand were an rvalue. Otherwise, or **if the overload resolution fails**, overload resolution is performed again, considering the expression or operand as an lvalue.